js/piwik.js

```diff
@@ -2855,6 +2872,11 @@ if (typeof window.Matomo !== 'object') {
                     callback({request: request, trackerUrl: configTrackerUrl, success: true, isSendBeacon: true});
                 }
 
+                // If the query parameter indicating this is a test exists, close after first request is sent
+                if (wasJsTrackingCodeInstallCheckParamProvided() && isObject(windowAlias)) {
+                    windowAlias.close();
+                }
+
```

Comment:
Have you added that only to sendBeacon on purpose? If sending a beacon isn't possible, due to browser support, or due to being disabled in the tracking code, this would never be executed. I guess the check should be performed for any sending method that is available, right?

Reply:
Thank you. That's a good point. I'll look for a better location for it.

Reply:
@sgiehl  I forgot that I tried putting it in the `sendXmlHttpRequest` function, but it was never executed. I guess I could leave it in `sendPostRequestViaSendBeacon` but also add it to the end of `sendXmlHttpRequest` again. Does that sound alright?

Reply:
I ended up also adding it to the `getImage` function as well, since that's what was used when I disabled the sendBeacon.